src/include/migraphx/op/concat.hpp

```diff
@@ -73,49 +74,103 @@ struct concat
         }
         return offsets;
     }
+
     shape normalize_compute_shape(std::vector<shape> inputs) const
     {
+        // Todo:: not using check_shapes struct; it
+        // doesn't have a method to check for at least one shape.
```

Comment:
It can be used `.has(1, 2)`.

Reply:
no, that won't work. It can have [1, inf] inputs.